Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -211,52 +472,88 @@ class GeneralForceSubsystemRep : public ForceSubsystem::Guts {
         // force element is enabled.
         const Array_<bool>& forceEnabled = Value< Array_<bool> >::downcast
                                     (getDiscreteVariable(s, forceEnabledIndex));
+        const Array_<bool>& parallelEnabled = Value< Array_<bool> >::downcast
+                                    (getDiscreteVariable(s, parallelEnabledIndex));
 
         // Get access to System-global force cache arrays.
-        Vector_<SpatialVec>&   rigidBodyForces = 
+        Vector_<SpatialVec>&   rigidBodyForces =
                                     mbs.updRigidBodyForces(s, Stage::Dynamics);
-        Vector_<Vec3>&         particleForces  = 
+        Vector_<Vec3>&         particleForces  =
                                     mbs.updParticleForces (s, Stage::Dynamics);
-        Vector&                mobilityForces  = 
+        Vector&                mobilityForces  =
                                     mbs.updMobilityForces (s, Stage::Dynamics);
 
+        // Track the enabled forces and whether they should be parallelized.
+        Array_<Force*> enabledNonParallelForces;
+        Array_<Force*> enabledParallelForces;
+
+        // Avoid repeatedly allocating memory.
+        enabledNonParallelForces.reserve(forces.size());
+        enabledParallelForces.reserve(forces.size());
+
+        for (int i = 0; i < (int) forces.size(); ++i) {
+            if(forceEnabled[i])
+            {
+                if (parallelEnabled[i])
+                    enabledParallelForces.push_back(forces[i]);
+                else
+                    enabledNonParallelForces.push_back(forces[i]);
+            }
+        }
+        // cout << "------------------------------------------------" << endl;
+        // cout << "Parallel Forces:" << endl;
+        // cout << enabledParallelForces << endl;
+        // cout << endl;
+        // cout << "NonParallel Forces" << endl;
+        // cout << enabledNonParallelForces << endl;
+        // cout << endl;
+        // cout << "rigidBodyForces" << endl;
+        // cout << rigidBodyForces << endl;
+        // cout << endl;
+        // cout << "particleForces" << endl;
+        // cout << particleForces << endl;
+        // cout << endl;
+        // cout << "mobilityForces" << endl;
+        // cout << mobilityForces << endl;
+        // cout << endl;
```

Comment:
Can this be removed (you can keep this locally, but not commit it to the git repo; git lets you add only sections of files)?